functions/src/controllers/sessions.ts

```diff
@@ -105,6 +105,10 @@ export const getSession = async (
     throw new RequestError(ValidateSessionError.userNotFound);
   }
 
+  if (!isUserAllowedToJoin(session, userId)) {
```

Comment:
This check and the one above does the same thing checking if the user is allowed to join. Maybe we only need one here? 
Sometimes we do this user check in the controller, and sometimes in the firestore query. Maybe we should do it the same way always? - maybe no in this PR, but as a next improvement?

Reply:
Yea, lump it together, user is in and it's private, or the time check. I'll release now and write a note